### PR TITLE
[DEV-1434]: fetch blocks content from Strapi

### PR DESCRIPTION
**Pull request by MikeAtUqido**
#### List of Changes

* Fetch Strapi's blocks content

#### Motivation and Context

The Strapi blocks content type didn't have a codec

#### How Has This Been Tested?

Manually and with unit tests

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by marcobottaro** [comment]
@MikeAtUqido the build fails due to this type error:
`Error: Invalid value undefined supplied to : {| data: {| attributes: {| comingsoonDocumentation: { title: string, links: Array<{| text: string, href: string, target: (NullToUndefined | "_self" | "_blank" | "_parent" | "_top") |}> }, heroSlider: Array<{| title: string, subhead: (NullToUndefined | Array<({| type: "paragraph", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "quote", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "code", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "heading", level: (1 | 2 | 3 | 4 | 5 | 6), children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | { type: "list", format: ("ordered" | "unordered"), children: Array<({| type: "list-item", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | ListBlockNodeCodec)> } | {| type: "image", image: ({| name: string, url: string, width: number, height: number, hash: string, ext: string, mime: string, size: number, provider: string, createdAt: string, updatedAt: string |} & Partial<{ alternativeText: string, caption: string, formats: { [K in ListBlockNodeCodec]: string }, previewUrl: string, provider_metadata: ListBlockNodeCodec }>), children: Array<{| type: "text", text: "" |}> |})>), callToAction: (NullToUndefined | {| link: {| text: string, href: string, target: (NullToUndefined | "_self" | "_blank" | "_parent" | "_top") |}, variant: (NullToUndefined | "text" | "contained" | "outlined") |}), titleColor: (NullToUndefined | "contrastText" | "main" | "light" | "dark"), backgroundImage: {| data: (NullToUndefined | {| attributes: {| name: string, alternativeText: (null | string), caption: (null | string), width: number, height: number, ext: string, mime: string, url: string |} |}) |} |}>, newsShowcase: (null | {| title: string, items: {| data: Array<{| attributes: {| comingSoon: boolean, title: string, link: {| text: string, href: string, target: (NullToUndefined | "_self" | "_blank" | "_parent" | "_top") |}, publishedAt: DateFromISOString, image: {| data: (null | {| attributes: {| name: string, alternativeText: (null | string), caption: (null | string), width: number, height: number, ext: string, mime: string, url: string |} |}) |} |} |}> |} |}), productsShowcase: {| title: string, products: {| data: Array<{| attributes: {| name: string, description: (null | string), slug: string, logo: {| data: {| attributes: {| name: string, alternativeText: (null | string), caption: (null | string), width: number, height: number, ext: string, mime: string, url: string |} |} |} |} |}> |} |} |} |} |}/data: {| attributes: {| comingsoonDocumentation: { title: string, links: Array<{| text: string, href: string, target: (NullToUndefined | "_self" | "_blank" | "_parent" | "_top") |}> }, heroSlider: Array<{| title: string, subhead: (NullToUndefined | Array<({| type: "paragraph", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "quote", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "code", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "heading", level: (1 | 2 | 3 | 4 | 5 | 6), children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | { type: "list", format: ("ordered" | "unordered"), children: Array<({| type: "list-item", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | ListBlockNodeCodec)> } | {| type: "image", image: ({| name: string, url: string, width: number, height: number, hash: string, ext: string, mime: string, size: number, provider: string, createdAt: string, updatedAt: string |} & Partial<{ alternativeText: string, caption: string, formats: { [K in ListBlockNodeCodec]: string }, previewUrl: string, provider_metadata: ListBlockNodeCodec }>), children: Array<{| type: "text", text: "" |}> |})>), callToAction: (NullToUndefined | {| link: {| text: string, href: string, target: (NullToUndefined | "_self" | "_blank" | "_parent" | "_top") |}, variant: (NullToUndefined | "text" | "contained" | "outlined") |}), titleColor: (NullToUndefined | "contrastText" | "main" | "light" | "dark"), backgroundImage: {| data: (NullToUndefined | {| attributes: {| name: string, alternativeText: (null | string), caption: (null | string), width: number, height: number, ext: string, mime: string, url: string |} |}) |} |}>, newsShowcase: (null | {| title: string, items: {| data: Array<{| attributes: {| comingSoon: boolean, title: string, link: {| text: string, href: string, target: (NullToUndefined | "_self" | "_blank" | "_parent" | "_top") |}, publishedAt: DateFromISOString, image: {| data: (null | {| attributes: {| name: string, alternativeText: (null | string), caption: (null | string), width: number, height: number, ext: string, mime: string, url: string |} |}) |} |} |}> |} |}), productsShowcase: {| title: string, products: {| data: Array<{| attributes: {| name: string, description: (null | string), slug: string, logo: {| data: {| attributes: {| name: string, alternativeText: (null | string), caption: (null | string), width: number, height: number, ext: string, mime: string, url: string |} |} |} |} |}> |} |} |} |}/attributes: {| comingsoonDocumentation: { title: string, links: Array<{| text: string, href: string, target: (NullToUndefined | "_self" | "_blank" | "_parent" | "_top") |}> }, heroSlider: Array<{| title: string, subhead: (NullToUndefined | Array<({| type: "paragraph", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "quote", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "code", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "heading", level: (1 | 2 | 3 | 4 | 5 | 6), children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | { type: "list", format: ("ordered" | "unordered"), children: Array<({| type: "list-item", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | ListBlockNodeCodec)> } | {| type: "image", image: ({| name: string, url: string, width: number, height: number, hash: string, ext: string, mime: string, size: number, provider: string, createdAt: string, updatedAt: string |} & Partial<{ alternativeText: string, caption: string, formats: { [K in ListBlockNodeCodec]: string }, previewUrl: string, provider_metadata: ListBlockNodeCodec }>), children: Array<{| type: "text", text: "" |}> |})>), callToAction: (NullToUndefined | {| link: {| text: string, href: string, target: (NullToUndefined | "_self" | "_blank" | "_parent" | "_top") |}, variant: (NullToUndefined | "text" | "contained" | "outlined") |}), titleColor: (NullToUndefined | "contrastText" | "main" | "light" | "dark"), backgroundImage: {| data: (NullToUndefined | {| attributes: {| name: string, alternativeText: (null | string), caption: (null | string), width: number, height: number, ext: string, mime: string, url: string |} |}) |} |}>, newsShowcase: (null | {| title: string, items: {| data: Array<{| attributes: {| comingSoon: boolean, title: string, link: {| text: string, href: string, target: (NullToUndefined | "_self" | "_blank" | "_parent" | "_top") |}, publishedAt: DateFromISOString, image: {| data: (null | {| attributes: {| name: string, alternativeText: (null | string), caption: (null | string), width: number, height: number, ext: string, mime: string, url: string |} |}) |} |} |}> |} |}), productsShowcase: {| title: string, products: {| data: Array<{| attributes: {| name: string, description: (null | string), slug: string, logo: {| data: {| attributes: {| name: string, alternativeText: (null | string), caption: (null | string), width: number, height: number, ext: string, mime: string, url: string |} |} |} |} |}> |} |} |}/heroSlider: Array<{| title: string, subhead: (NullToUndefined | Array<({| type: "paragraph", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "quote", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "code", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "heading", level: (1 | 2 | 3 | 4 | 5 | 6), children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | { type: "list", format: ("ordered" | "unordered"), children: Array<({| type: "list-item", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | ListBlockNodeCodec)> } | {| type: "image", image: ({| name: string, url: string, width: number, height: number, hash: string, ext: string, mime: string, size: number, provider: string, createdAt: string, updatedAt: string |} & Partial<{ alternativeText: string, caption: string, formats: { [K in ListBlockNodeCodec]: string }, previewUrl: string, provider_metadata: ListBlockNodeCodec }>), children: Array<{| type: "text", text: "" |}> |})>), callToAction: (NullToUndefined | {| link: {| text: string, href: string, target: (NullToUndefined | "_self" | "_blank" | "_parent" | "_top") |}, variant: (NullToUndefined | "text" | "contained" | "outlined") |}), titleColor: (NullToUndefined | "contrastText" | "main" | "light" | "dark"), backgroundImage: {| data: (NullToUndefined | {| attributes: {| name: string, alternativeText: (null | string), caption: (null | string), width: number, height: number, ext: string, mime: string, url: string |} |}) |} |}>/0: {| title: string, subhead: (NullToUndefined | Array<({| type: "paragraph", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "quote", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "code", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "heading", level: (1 | 2 | 3 | 4 | 5 | 6), children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | { type: "list", format: ("ordered" | "unordered"), children: Array<({| type: "list-item", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | ListBlockNodeCodec)> } | {| type: "image", image: ({| name: string, url: string, width: number, height: number, hash: string, ext: string, mime: string, size: number, provider: string, createdAt: string, updatedAt: string |} & Partial<{ alternativeText: string, caption: string, formats: { [K in ListBlockNodeCodec]: string }, previewUrl: string, provider_metadata: ListBlockNodeCodec }>), children: Array<{| type: "text", text: "" |}> |})>), callToAction: (NullToUndefined | {| link: {| text: string, href: string, target: (NullToUndefined | "_self" | "_blank" | "_parent" | "_top") |}, variant: (NullToUndefined | "text" | "contained" | "outlined") |}), titleColor: (NullToUndefined | "contrastText" | "main" | "light" | "dark"), backgroundImage: {| data: (NullToUndefined | {| attributes: {| name: string, alternativeText: (null | string), caption: (null | string), width: number, height: number, ext: string, mime: string, url: string |} |}) |} |}/subhead: (NullToUndefined | Array<({| type: "paragraph", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "quote", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "code", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | {| type: "heading", level: (1 | 2 | 3 | 4 | 5 | 6), children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | { type: "list", format: ("ordered" | "unordered"), children: Array<({| type: "list-item", children: Array<(({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>) | {| type: "link", url: string, children: Array<({| type: "text", text: string |} & Partial<{ bold: boolean, italic: boolean, underline: boolean, strikethrough: boolean, code: boolean }>)> |})> |} | ListBlockNodeCodec)> } | {| type: "image", image: ({| name: string, url: string, width: number, height: number, hash: string, ext: string, mime: string, size: number, provider: string, createdAt: string, updatedAt: string |} & Partial<{ alternativeText: string, caption: string, formats: { [K in ListBlockNodeCodec]: string }, previewUrl: string, provider_metadata: ListBlockNodeCodec }>), children: Array<{| type: "text", text: "" |}> |})>)/0: NullToUndefined`

File: apps/nextjs-website/src/app/page.tsx [hunk 1]
```diff
@@ -41,12 +39,11 @@ const Home = async () => {
       <NotSsrWebinarHeaderBanner webinars={webinars} />
 
       <HeroSwiper
-        cards={homepage.hero.map((itemProp, index) => ({
+        cards={homepage.hero.map((itemProp) => ({
```
Comment (marcobottaro) [inline apps/nextjs-website/src/app/page.tsx:42]: Shouldn't each child in a list have a unique "key" prop?
```suggestion
        cards={homepage.hero.map((itemProp, index) => ({
```

File: apps/nextjs-website/src/app/page.tsx [hunk 2]
```diff
@@ -41,12 +39,11 @@ const Home = async () => {
...
           ...itemProp,
-          child:
-            index === 0 ? (
-              <SiteLabel title={header.title} boldTitle={header.boldTitle} />
-            ) : undefined,
+          child: itemProp.subhead && (
+            <BlocksRendererClient content={itemProp.subhead} />
```
Comment (marcobottaro) [inline apps/nextjs-website/src/app/page.tsx:45]: As a consequence, this line would become
```suggestion
            <BlocksRendererClient key={index} content={itemProp.subhead} />
```

File: apps/nextjs-website/src/components/atoms/CtaSlide/CtaSlide.tsx [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
 import { Box, Stack, Typography, useTheme } from '@mui/material';
 import { ButtonNaked } from '@pagopa/mui-italia';
+import { BlocksContent } from '@strapi/blocks-react-renderer';
 import Image from 'next/image';
 import Link from 'next/link';
 import React, { ReactNode } from 'react';
 
 export type CtaSlideProps = {
   readonly title: string;
+  readonly subhead?: BlocksContent | null;
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/atoms/CtaSlide/CtaSlide.tsx:11]: Is `null` value a real option? Didn't you used `NullToUndefined` type in `HeroSlideCodec` constant?

File: apps/nextjs-website/src/components/molecules/BlocksRendererClient/BlocksRendererClient.tsx [hunk 4]
```diff
@@ -0,0 +1,35 @@
+'use client';
+import { Typography } from '@mui/material';
+import { BlocksContent, BlocksRenderer } from '@strapi/blocks-react-renderer';
+import Image from 'next/image';
+
+const BlocksRendererClient = ({
+  content,
+}: {
+  readonly content: BlocksContent | undefined | null;
```
Comment (marcobottaro) [inline apps/nextjs-website/src/components/molecules/BlocksRendererClient/BlocksRendererClient.tsx:9]: Same here

File: apps/nextjs-website/src/lib/strapi/__tests__/homepage.test.ts [hunk 5]
```diff
@@ -48,6 +49,42 @@ const makeStrapiResponseJson = () => ({
...
+                  text: 'aText',
+                  type: 'text',
+                  strikethrough: true,
+                },
+                {
+                  text: 'aText',
+                  type: 'text',
+                  code: true,
+                },
+              ],
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/__tests__/homepage.test.ts:85]: Maybe you should add  paragraph and  heading types to this test

File: apps/nextjs-website/src/lib/strapi/codecs/NullToUndefined.ts
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/codecs/NullToUndefined.ts:1]: Weel done :+1:

File: apps/nextjs-website/src/lib/strapi/fetchFromStrapi.ts [hunk 6]
```diff
@@ -28,6 +28,7 @@ export const fetchFromStrapi = <A, O, I>(
             () =>
               fetchFun(`${strapiEndpoint}/api/${path}/?${populate}`, {
                 method: 'GET',
+                cache: 'no-store',
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/fetchFromStrapi.ts:31]: I think you should remove this change from this PR.
This option is useful for local development, but useless in production and anyhow is not part of this task

File: apps/nextjs-website/src/lib/strapi/codecs/NullToUndefined.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+/**
+ * This type is used to convert null to undefined when decoding.
+ * This is useful becuase Strapi returns null for missing fields, so we want to convert those to undefined to be more consistent with the rest of our codebase.
+ *
+ * @example
+ * const codec = t.union([NullToUndefined, t.string]);
+ * const result = codec.decode(null);
+ * // result will be Right(undefined)
+ */
+export const NullToUndefined = new t.Type<undefined, null, unknown>(
```
Comment (marcobottaro) [inline apps/nextjs-website/src/lib/strapi/codecs/NullToUndefined.ts:14]: Maybe we should rename this constant (and the file) to `NullToUndefinedCodec` to be consistent with `BlocksContentCodec`. @MarcoPonchia @tommaso1 @datalek @kin0992 What do you think about this?

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: apps/nextjs-website/src/lib/strapi/codecs/NullToUndefined.ts (hunk 7 above)
Comment (datalek) [reply apps/nextjs-website/src/lib/strapi/codecs/NullToUndefined.ts:14]: Given that for me it's fine both with and without Codec. Having said that, I see the Codec suffix a good fit for complex types (e.g. structures). I see `NullToUndefined` more like [DateFromISOString](https://github.com/gcanti/io-ts-types/blob/b3a5616/src/DateFromISOString.ts#L24)

File: apps/nextjs-website/src/lib/strapi/codecs/BlocksContentCodec.ts [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    type: t.literal('text'),
+    text: t.string,
+  }),
+  t.partial({
+    bold: t.boolean,
+    italic: t.boolean,
+    underline: t.boolean,
+    strikethrough: t.boolean,
+    code: t.boolean,
+  }),
```
Comment (datalek) [inline apps/nextjs-website/src/lib/strapi/codecs/BlocksContentCodec.ts:15]: I can't find which use case the `partial` covers. If I'm not wrong, Strapi always returns properties, even if a property is not populated. In such cases, Strapi returns the property evaluated to null

File: apps/nextjs-website/src/lib/strapi/codecs/BlocksContentCodec.ts [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    t.union([
+      ListItemInlineNodeCodec,
+      t.recursion('ListBlockNodeCodec', () => t.unknown),
+    ])
+  ),
+});
+
+const ImageBlockChildCodec = t.strict({
+  type: t.literal('text'),
+  text: t.literal(''),
```
Comment (datalek) [inline apps/nextjs-website/src/lib/strapi/codecs/BlocksContentCodec.ts:76]: I'm a little confused by this line of code, maybe it is a typo. If it is not, could you explain it to me?

**Review by datalek — COMMENTED** [review]
I left just a couple of comments 🙂

File: apps/nextjs-website/src/lib/strapi/codecs/BlocksContentCodec.ts (hunk 8 above)
Comment (MikeAtUqido) [reply apps/nextjs-website/src/lib/strapi/codecs/BlocksContentCodec.ts:15]: That's not the case. If the type returned by Strapi is `blocks`, then these fields could be populated or not.
This is an example returned by Strapi:
```json
"subhead": [
            {
              "type": "paragraph",
              "children": [
                { "type": "text", "text": "Ciao " },
                { "text": "ciao ", "type": "text", "bold": true },
                { "text": "ciao ", "type": "text", "italic": true },
                { "text": "ciao", "type": "text", "underline": true },
                { "text": " ", "type": "text" },
                { "text": "ciao", "type": "text", "strikethrough": true },
                { "text": " ", "type": "text" },
                { "text": "ciao", "type": "text", "code": true },
                { "text": " ciao", "type": "text" }
              ]
            }
          ],
```

File: apps/nextjs-website/src/lib/strapi/codecs/BlocksContentCodec.ts (hunk 9 above)
Comment (MikeAtUqido) [reply apps/nextjs-website/src/lib/strapi/codecs/BlocksContentCodec.ts:76]: This codec is a "translations" of the types coming from the [blocks-react-renderer](https://github.com/strapi/blocks-react-renderer/tree/main) library developed by Strapi to render the `blocks` content types, and this line comes directly from [there](https://github.com/strapi/blocks-react-renderer/blob/main/src/BlocksRenderer.tsx#L74)

**Comment by datalek** [comment]
> @MikeAtUqido the build fails due to this type error: `Error: Invalid value undefined supplied to : {| data: {| attributes: ... NullToUndefined` Take a look here: https://github.com/pagopa/developer-portal/actions/runs/8279763676/job/22654946657?pr=745

I opened a PR #770 that try to improve the readability of this kind of errors 🤞

File: apps/strapi-cms/src/components/common/hero-slide.json [hunk 10]
```diff
@@ -9,6 +9,10 @@
     "title": {
       "type": "string"
     },
+    "subhead" : {
+      "type": "blocks",
+      "required": false
+    },
```
Comment (marcobottaro) [inline apps/strapi-cms/src/components/common/hero-slide.json:15]: To avoid build errors, I think you should separate this PR. This would consent us to merge the changes to Strapi before the FE pull request

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (marcobottaro) [inline package-lock.json:1]: Are these changes required?

**Review by marcobottaro — CHANGES_REQUESTED** [review]

File: .changeset/fair-feet-exercise.md [hunk 11]
```diff
@@ -0,0 +1,5 @@
+---
+"strapi-cms": minor
+---
+
+Fetch Strapi's blocks content type
```
Comment (datalek) [inline .changeset/fair-feet-exercise.md:5]:
```suggestion
Add subhead to hero slide
```

**Review by datalek — CHANGES_REQUESTED** [review]

**Review by datalek — APPROVED** [review]
LGTM

**Review by marcobottaro — APPROVED** [review]

**Review by tommaso1 — APPROVED** [review]

**Review by MarcoPonchia — APPROVED** [review]

**Review by MarcoPonchia — APPROVED** [review]

**Review by marcobottaro — APPROVED** [review]

**Review by marcobottaro — APPROVED** [review]

**Review by MarcoPonchia — APPROVED** [review]
